lib/mail/message.rb

```diff
@@ -1503,8 +1503,10 @@ def charset
     # Sets the charset to the supplied value.
     def charset=(value)
       @defaulted_charset = false
-      @charset = value
-      @header.charset = value
+      if value
+        @charset = value
+        @header.charset = value
+      end
```

Comment:
How would someone remove the charset now, to rely on default parsing?

This is seeming like something to fix downstream in Action Mailer.

Reply:
it isn't just "bad tests", it breaks Action Mailer for real. Even if it'll be fixed in Action Mailer, Mail 2.8 will be incompatible with all previous Rails versions, I think we should avoid that for all costs.

what about skipping blank parameters? https://github.com/mikel/mail/blob/master/lib/mail/fields/parameter_hash.rb#L47

Reply:
ok, I fixed that by removing the existing header if you set charset = nil

note that this case didn't work even on previous mail versions including 2.7.1 stable